lib/throttler.js

```diff
@@ -10,86 +11,107 @@ var defaults      = require('../defaults'),
  * @param {object} config
  * @param {function} limited will be called upon limit reached
  */
-module.exports = function create_throttler(config, limited) {
-    if (typeof config == 'function') {
-        return create_throttler(null, config);
+module.exports = function createThrottler (config, limited) {
+  // createThrottler()
+  if (!config) {
+    config = {}
+  }
+
+  // createThrottler(fn)
+  if (typeof config === 'function') {
+    limited = config
+    config = null
+  }
+
+  // createThrottler(config, false)
+  if (typeof limited !== 'function') {
+    limited = null
+  }
+  defaultsDeep(config, defaults)
+
+  var Throttle = throttleModel(config)
+  return throttleMiddleware
+
+  function errHandler (res, next, error) {
+    // DO WE WANT TO BREAK THE APP AND CAUSE A ISSUE OR DO WE WANT TO SILENTLY INTO THE NIGHT AND JUST LOG OUR ERRORS
```

Comment:
I think adding an emailer may be a bit much, but we can either:
- Add a flag to optionally fail silently
- Add an optional error handler